auth-server/src/services/email.ts

```diff
@@ -109,3 +111,40 @@ export const sendPostSubscriptionMail = (user: User, author: User, comment) => {
 	sgMail.send(msg).catch(e =>
 		console.error('Post subscription email not sent', e));
 };
+
+export const sendEmailUndoEmail = (user: User, undoToken: EmailUndoToken) => {
+	if (!apiKey) {
+		console.warn('Email undo token email not sent due to missing API key');
+		return;
+	}
+
+	const undoUrl = `${DOMAIN}/undo-email/${undoToken.token}`;
+	const text = `
+		<p>
+			Hi ${user.name || ''}!<br/><br/>
+
+			It looks like someone has changed your email at polkassembly.<br />
+			If its you then its ok.<br /><br />
+
+			If its not you then here is a recovery email link. Use it to change your account email back to this email:<br /><br />
+			<a href="${undoUrl}">Recover Your Email</a><br /><br />
+
+			Just a heads up, to make sure your information is safe and secure, the link will expire after 24 hours.<br /><br />
```

Comment:
I did not see this in the code. We should give probably a little more time, thinking about 48h.

```suggestion
			This link is valid for 48 hours, past this time, you will not be able to use it to recover your email. If you did not have time to click it and are a victim of a malicious email change, please open an issue on https://github.com/paritytech/polkassembly/issues/new<br /><br />
```